recipes/libmikmod/all/conanfile.py

```diff
@@ -0,0 +1,123 @@
+            if self.options.with_alsa:
+                self.requires("libalsa/1.1.9")
+            if self.options.with_pulse:
+                self.requires("pulseaudio/13.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self, set_cmake_flags=True)
+        cmake.definitions["ENABLE_STATIC"] = not self.options.shared
+        cmake.definitions["ENABLE_DOC"] = False
+        cmake.definitions["ENABLE_DSOUND"] = self.options.get_safe("with_dsound", False)
+        cmake.definitions["ENABLE_MMSOUND"] = self.options.get_safe("with_mmsound", False)
+        cmake.definitions["ENABLE_ALSA"] = self.options.get_safe("with_alsa", False)
+        cmake.definitions["ENABLE_OSS"] = self.options.get_safe("with_oss", False)
+        cmake.definitions["ENABLE_PULSE"] = self.options.get_safe("with_pulse", False)
+        cmake.definitions["ENABLE_COREAUDIO"] = self.options.get_safe("with_coreaudio", False)
+        cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        # 0001:
+        #   Patch CMakeLists.txt to run `conan_basic_setup`, to avoid building shared lib when
+        #   shared=False, and a fix to install .dlls correctly on Windows
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+         # Ensure missing dependencies yields errors
+        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                              "MESSAGE(WARNING",
+                              "MESSAGE(FATAL_ERROR")
+
+        tools.replace_in_file(os.path.join(self._source_subfolder, "drivers", "drv_alsa.c"),
+                              "alsa_pcm_close(pcm_h);",
+                              "if (pcm_h) alsa_pcm_close(pcm_h);")
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING.LESSER", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        os.remove(os.path.join(self.package_folder, "bin", "libmikmod-config"))
+        if not self.options.shared:
+            tools.rmdir(os.path.join(self.package_folder, "bin"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if not self.options.shared:
+            self.cpp_info.defines = ["MIKMOD_STATIC"]
+
+        if self.options.get_safe("with_dsound"):
+            self.cpp_info.system_libs.append("dsound")
+        if self.options.get_safe("with_mmsound"):
+            self.cpp_info.system_libs.append("winmm")
+        if self.options.get_safe("with_coreaudio"):
```

Comment:
Is this mac only?

Reply:
yes, but this option is deleted on non macos platforms